pytorch_lightning/strategies/ipu.py

```diff
@@ -214,10 +214,11 @@ def lightning_module(self) -> Optional["pl.LightningModule"]:
     def _convert_to_poptorch_loader(
         self, dataloader: DataLoader, sampler, mode: Optional[RunningStage] = None
     ) -> "poptorch.DataLoader":
-        dl_kwargs = _get_dataloader_init_kwargs(dataloader, sampler)
-        # Override to drop last uneven batch, as IPUs does not support uneven inputs.
-        dl_kwargs["drop_last"] = True
+        if isinstance(dataloader, poptorch.DataLoader):
+            # the user is returning the `poptorch.DataLoader` directly, don't change anything.
+            return dataloader
```

Comment:
This is not strictly related to the bugfix, but we probably should not recreate the dataloader if the user already created a `poptorch.DataLoader`